apriltag_ros:
  ros__parameters:
    # AprilTag 3 code parameters
    # Find descriptions in apriltag/include/apriltag.h:struct apriltag_detector
    #                      apriltag/include/apriltag.h:struct apriltag_family
    tag_family: 'tagStandard41h12' # options: tagStandard52h13, tagStandard41h12, tag36h11, tag25h9, tag16h5, tagCustom48h12, tagCircle21h7, tagCircle49h12
    tag_threads: 2          # default: 2
    tag_decimate: 1.0        # default: 1.0
    tag_blur:          0.0        # default: 0.0
    tag_refine_edges:  1          # default: 1
    tag_debug:         0          # default: 0
    max_hamming_dist:  2          # default: 2 (Tunable parameter with 2 being a good choice - values >=3 consume large amounts of memory. Choose the largest value possible.)
    # Other parameters
    publish_tf:        false       # default: false
    transport_hint:    "raw"      # default: raw, see http://wiki.ros.org/image_transport#Known_Transport_Packages for options

    # # Definitions of tags to detect
    #
    # ## General remarks
    #
    # - All length in meters
    # - 'size' refers to the length of the shared border between solid black and solid white rectangle.
    #   See README.md or https://github.com/AprilRobotics/apriltag/wiki/AprilTag-User-Guide#pose-estimation for details.
    # - Ellipsis (...) signifies that the previous element can be repeated multiple times.
    #
    # ## Standalone tag definitions
    # ### Remarks
    #
    # - name is optional
    #
    # ### Syntax
    #
    # ROS 1 PARAM
    # standalone_tags:
    #   [
    #     {id: ID, size: SIZE, name: NAME},
    #     ...
    #   ]

    #ROS 2 PARAM
    standalone_tags_num: 0
    #standalone_tags_X:
    #  id: ID
    #  size: SIZE e.g. 0.036717255 ----> 3.6 cm by 3.6 cm
    #  name: 'NAME'




    # ## Tag bundle definitions
    # ### Remarks
    #
    # - name is optional
    # - x, y, z have default values of 0 thus they are optional
    # - qw has default value of 1 and qx, qy, qz have default values of 0 thus they are optional
    #
    # ### Syntax
    #
    tag_bundles_num: 3
    #ROS 2 PARAM
    #tag_bundles_X:
    #  name: 'CUSTOM_BUNDLE_NAME'
    #  layout_num: 0
    #  layout_X:
    #    id: ID
    #    size: SIZE e.g. 0.036717255 ----> 3.6 cm by 3.6 cm
    #    x: X_POS
    #    y: Y_POS
    #    z: Z_POS
    #    qx: QUAT_X_VAL
    #    qy: QUAT_Y_VAL
    #    qz: QUAT_Z_VAL
    #    qw: QUAT_W_VAL
    tag_bundles_0:
      name: 'landingpad_004'
      layout_num: 5
      layout_0:
        id: 0
        size: 0.036717255
        x: 0.0
        y: 0.0
        z: 0.0
        qx: 0.0
        qy: 0.0
        qz: 0.0
        qw: 1.0
      layout_1:
        id: 1
        size: 0.075585559
        x: -0.23
        y: 0.0
        z: 0.0
        qx: 0.0
        qy: 0.0
        qz: 0.0
        qw: 1.0
      layout_2:
        id: 2
        size: 0.075585559
        x: 0.23
        y: 0.0
        z: 0.0
        qx: 0.0
        qy: 0.0
        qz: 0.0
        qw: 1.0
      layout_3:
        id: 3
        size: 0.075585559
        x: 0.0
        y: 0.23
        z: 0.0
        qx: 0.0
        qy: 0.0
        qz: 0.0
        qw: 1.0
      layout_4:
        id: 4
        size: 0.075585559
        x: 0.0
        y: -0.23
        z: 0.0
        qx: 0.0
        qy: 0.0
        qz: 0.0
        qw: 1.0
    tag_bundles_1:
      name: 'nest_marker_001'
      layout_num: 5
      layout_0:
        id: 6
        size: 0.3385
        x: 0.0
        y: 0.0
        z: 0.0
        qx: 0.0
        qy: 0.0
        qz: 0.0
        qw: 1.0
      layout_1:
        id: 7
        size: 0.16667
        x: 0.0
        y: -0.46
        z: 0.0
        qx: 0.0
        qy: 0.0
        qz: 0.0
        qw: 1.0
      layout_2:
        id: 8
        size: 0.16667
        x: 0.0
        y: 0.45
        z: 0.0
        qx: 0.0
        qy: 0.0
        qz: 0.0
        qw: 1.0
      layout_3:
        id: 9
        size: 0.16667
        x: -0.46
        y: 0.0
        z: 0.0
        qx: 0.0
        qy: 0.0
        qz: 0.0
        qw: 1.0
      layout_4:
        id: 10
        size: 0.16667
        x: 0.46
        y: 0.0
        z: 0.0
        qx: 0.0
        qy: 0.0
        qz: 0.0
        qw: 1.0
    tag_bundles_2:
      name: 'nest_marker_002'
      layout_num: 5
      layout_0:
        id: 11
        size: 0.3385
        x: 0.0
        y: 0.0
        z: 0.0
        qx: 0.0
        qy: 0.0
        qz: 0.0
        qw: 1.0
      layout_1:
        id: 12
        size: 0.16667
        x: 0.0
        y: -0.46
        z: 0.0
        qx: 0.0
        qy: 0.0
        qz: 0.0
        qw: 1.0
      layout_2:
        id: 13
        size: 0.16667
        x: 0.0
        y: 0.45
        z: 0.0
        qx: 0.0
        qy: 0.0
        qz: 0.0
        qw: 1.0
      layout_3:
        id: 14
        size: 0.16667
        x: -0.46
        y: 0.0
        z: 0.0
        qx: 0.0
        qy: 0.0
        qz: 0.0
        qw: 1.0
      layout_4:
        id: 15
        size: 0.16667
        x: 0.46
        y: 0.0
        z: 0.0
        qx: 0.0
        qy: 0.0
        qz: 0.0
        qw: 1.0

    #ROS 1 PARAM
    # tag_bundles:
    #   [
    #     {
    #       name: 'CUSTOM_BUNDLE_NAME',
    #       layout:
    #         [
    #           {id: ID, size: SIZE, x: X_POS, y: Y_POS, z: Z_POS, qw: QUAT_W_VAL, qx: QUAT_X_VAL, qy: QUAT_Y_VAL, qz: QUAT_Z_VAL},
    #           ...
    #         ]
    #     },
    #     ...
    #   ]